### PR TITLE
Add Hash extensions to ActiveSupport

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

**Comment by manhhung741** [comment]
I think the return type T.self_type on those methods are incorrect.

File: lib/activesupport/all/activesupport.rbi [hunk 1]
```diff
@@ -417,3 +417,35 @@ module ActiveSupport::NumberHelper
...
+  sig { returns(T.self_type) }
+  def deep_symbolize_keys!; end
+
+  sig { returns(T.self_type) }
+  def deep_transform_keys; end
+
+  sig { returns(T.self_type) }
+  def deep_transform_keys!; end
+
+  sig { returns(T.self_type) }
```
Comment (manhhung741) [inline lib/activesupport/all/activesupport.rbi]: This should change the key of a Hash from any -> string. `T.self_type` is wrong

File: lib/activesupport/all/activesupport.rbi [hunk 2]
```diff
@@ -417,3 +417,35 @@ module ActiveSupport::NumberHelper
...
+  def stringify_keys; end
+
+  sig { returns(T.self_type) }
+  def stringify_keys!; end
+
+  sig { returns(T.self_type) }
```
Comment (manhhung741) [inline lib/activesupport/all/activesupport.rbi]: This should change the key of a hash from any -> symbol. `T.self_type` is incorrect

File: lib/activesupport/all/activesupport.rbi (hunk 1 above)
Comment (frewsxcv) [reply lib/activesupport/all/activesupport.rbi]: ah you're right. are you able to open a pr?

**Comment by manhhung741** [comment]
I haven’t found a good solution yet. Ideally if this is on a T::Hash we can
change the type of the Key and keep the type of the value. But this is on a
Hash.

My temporary patch is to remove these methods.

**Comment by frewsxcv** [comment]
> My temporary patch is to remove these methods.

Why would we remove these methods? Why wouldn't we just use `T.untyped`?

**Comment by manhhung741** [comment]
It would be the same thing. It’s for my local copy so it doesnt really
matter.

**Comment by manhhung741** [comment]
So this actually works. I'll send out a PR later!


class Hash < Object
extend T::Sig

sig { returns(T::Hash[Symbol, V]) }
def symbolize_keys
T.unsafe(self)
end
end

**Comment by ghiculescu** [comment]
@manhhung741 what results did you get there? Maybe I'm misunderstanding, but I was hoping that in this example, the type of the value would be retained, rather than reverting to `T.untyped`.

```ruby
# hash.rbi

# typed: strict

class Hash
  extend T::Sig
  sig { returns(T::Hash[String, V]) }
  def stringify_keys; end
end
```

```ruby
# test.rb

# typed: true
require 'active_support/all'
require 'sorbet-runtime'

extend T::Sig

sig { returns( {a: String} ) }
def get_specific_hash_shape
  {a: "1"}
end

T.reveal_type(get_specific_hash_shape) # Revealed type: {a: String}
T.reveal_type(get_specific_hash_shape.stringify_keys) # T::Hash[String, T.untyped]

```

**Comment by ghiculescu** [comment]
Also @manhhung741 FYI on https://github.com/sorbet/sorbet-typed/pull/148

**Pull request merged**